atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheImpl.java

```diff
@@ -97,6 +98,28 @@ public synchronized CommitUpdate getCommitUpdate(long startTs) {
         return eventLog.getEventsBetweenVersions(versionBounds).toCommitUpdate(startVersion.get(), commitInfo);
     }
 
+    @Override
+    public SpanningCommitUpdate getSpanningCommitUpdate(long startTs) {
```

Comment:
Again, somewhat duplicated; can we deduplicate this? Not sure.

Also, I don't want `getCommitUpdate` to necessarily call this as it does more work than it needs to.

Reply:
If you want to deduplicate: you should probably just have a single method on `ClientLogEvents` that takes all three arguments, then call it with `commitInfo.commitVersion().version()` above and get the `CommitUpdate` from the returned spanning update.
Not entirely sure without seeing it if it's a bit too magical, though

Reply:
Gone for the most simple, which is to just use spanning commit updates and take the relevant parts only